app/src/androidTest/java/ch/epfl/skysync/end_to_end/E2EAddFlights.kt

```diff
@@ -0,0 +1,103 @@
+  private val repository = Repository(db)
+
+  @Before
+  fun setUpNavHost() {
+    composeTestRule.setContent {
+      navController = TestNavHostController(LocalContext.current)
+      navController.navigatorProvider.addNavigator(ComposeNavigator())
+      NavHost(navController = navController, startDestination = Route.MAIN) {
+        homeGraph(repository, navController, null)
+      }
+    }
+  }
+
+  @Test
+  fun addFlightAsAdmin() {
+
+    composeTestRule.onNodeWithTag("addFlightButton").performClick()
+    var route = navController.currentBackStackEntry?.destination?.route
+    Assert.assertEquals(Route.ADD_FLIGHT, route)
+    composeTestRule
+        .onNodeWithTag("Flight Lazy Column")
+        .performScrollToNode(hasTestTag("nb Passenger"))
+    composeTestRule.onNodeWithTag("nb Passenger").performClick()
+    composeTestRule.onNodeWithTag("nb Passenger").performTextClearance()
+    composeTestRule.onNodeWithTag("nb Passenger").performTextInput("13")
+
+    composeTestRule
+        .onNodeWithTag("Flight Lazy Column")
+        .performScrollToNode(hasTestTag("Date Field"))
+    composeTestRule.onNodeWithTag("Date Field").performClick()
+    composeTestRule.onNodeWithText("OK").performClick()
+
+    composeTestRule
+        .onNodeWithTag("Flight Lazy Column")
+        .performScrollToNode(hasTestTag("Flight Type Menu"))
+    composeTestRule.onNodeWithTag("Flight Type Menu").performClick()
+    composeTestRule.waitForIdle()
+    composeTestRule.onNodeWithTag("Flight Type 1").performClick()
+
+    composeTestRule
+        .onNodeWithTag("Flight Lazy Column")
+        .performScrollToNode(hasTestTag("Vehicle 0 Menu"))
+    composeTestRule.onNodeWithTag("Vehicle 0 Menu").performClick()
+    composeTestRule.onNodeWithTag("Vehicle 0 1").performClick()
+
+    composeTestRule
+        .onNodeWithTag("Flight Lazy Column")
+        .performScrollToNode(hasTestTag("Time Slot Menu"))
+    composeTestRule.onNodeWithTag("Time Slot Menu").performClick()
+    composeTestRule.onNodeWithTag("Time Slot 1").performClick()
+
+    composeTestRule
+        .onNodeWithTag("Flight Lazy Column")
+        .performScrollToNode(hasTestTag("Balloon Menu"))
+    composeTestRule.onNodeWithTag("Balloon Menu").performClick()
+    composeTestRule.onNodeWithTag("Balloon 1").performClick()
+
+    composeTestRule
+        .onNodeWithTag("Flight Lazy Column")
+        .performScrollToNode(hasTestTag("Basket Menu"))
+    composeTestRule.onNodeWithTag("Basket Menu").performClick()
+    composeTestRule.onNodeWithTag("Basket 1").performClick()
+    val title1 = "Add Flight"
+    composeTestRule.onNodeWithTag("$title1 Button").performClick()
+    route = navController.currentBackStackEntry?.destination?.route
+    Assert.assertEquals(Route.HOME, route)
+    var flightIsCreated = false
+    runTest {
+      val flights = repository.flightTable.getAll(onError = { assertNull(it) })
+      flightIsCreated = flights.any { it.nPassengers == 13 }
```

Comment:
this will cause the test to always pass if there's actually a flight with 13 passengers

Reply:
iirc it is initialised at test run so the db is empty.